### PR TITLE
Bug 1799332 - Add support for signing gecko frameworks in iscript

### PR DESCRIPTION
**Pull request by hneiva**

### CODE REVIEW

File: iscript/src/iscript/mac.py [hunk 1]
```diff
@@ -302,6 +302,13 @@ async def sign_app(sign_config, app_path, entitlements_path, provisioning_profil
                 continue
             if dir_.endswith((".app", ".appex")):
                 await sign_app(sign_config, abs_dir, entitlements_path)
+            if dir_.endswith(".framework"):
```
Comment (bhearsum) [inline iscript/src/iscript/mac.py:305]: Is there a reason you can't use `sign_app` here? AFAICT, this will cause the Framework to get signed without entitlements. I suspect if you add `.framework` to all the places (like the line above) that currently look for `.app` and `.appex`, that you may not need this block.

**Review by bhearsum — CHANGES_REQUESTED** [review]

File: iscript/src/iscript/mac.py (hunk 1 above)
Comment (hneiva) [reply iscript/src/iscript/mac.py:305]: I would fail at `app_executable = get_bundle_executable(app_path)` since Frameworks are not bundles:
```
2023-08-14 18:59:18,775 - scriptworker_client.utils - INFO - Running ['xattr', '-cr', 'ChannelPrefs.framework'] in /builds/scriptworker/work/0/Firefox Nightly.app/Contents/Frameworks ...
2023-08-14 18:59:18,913 - scriptworker_client.utils - INFO - ['xattr', '-cr', 'ChannelPrefs.framework'] in /builds/scriptworker/work/0/Firefox Nightly.app/Contents/Frameworks exited 0
2023-08-14 18:59:18,913 - iscript.mac - DEBUG - sign_app: signing ChannelPrefs.framework
Traceback (most recent call last):
  File "/builds/scriptworker/virtualenv/bin/iscript", line 33, in <module>
    sys.exit(load_entry_point('iscript', 'console_scripts', 'iscript')())
  File "/builds/checkouts/scriptworker-scripts/iscript/src/iscript/script.py", line 126, in main
    return sync_main(async_main, default_config=get_default_config())
  File "/builds/scriptworker/scriptworker-scripts/scriptworker_client/src/scriptworker_client/client.py", line 123, in sync_main
    loop.run_until_complete(_handle_asyncio_loop(async_main, config, task))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/builds/scriptworker/scriptworker-scripts/scriptworker_client/src/scriptworker_client/client.py", line 178, in _handle_asyncio_loop
    await async_main(config, task)
  File "/builds/checkouts/scriptworker-scripts/iscript/src/iscript/script.py", line 100, in async_main
    await func(config, task, **args)
  File "/builds/checkouts/scriptworker-scripts/iscript/src/iscript/mac.py", line 1401, in sign_and_pkg_behavior
    await sign_all_apps(config, sign_config, entitlements_path, all_paths, provisioning_profile_path)
  File "/builds/checkouts/scriptworker-scripts/iscript/src/iscript/mac.py", line 656, in sign_all_apps
    await raise_future_exceptions(futures)
  File "/builds/scriptworker/scriptworker-scripts/scriptworker_client/src/scriptworker_client/aio.py", line 56, in raise_future_exceptions
    raise exceptions[0]
  File "/builds/checkouts/scriptworker-scripts/iscript/src/iscript/mac.py", line 304, in sign_app
    await sign_app(sign_config, abs_dir, entitlements_path)
  File "/builds/checkouts/scriptworker-scripts/iscript/src/iscript/mac.py", line 284, in sign_app
    app_executable = get_bundle_executable(app_path)
  File "/builds/checkouts/scriptworker-scripts/iscript/src/iscript/mac.py", line 159, in get_bundle_executable
    return plistlib.readPlist(os.path.join(appdir, "Contents", "Info.plist"))["CFBundleExecutable"]
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/plistlib.py", line 99, in readPlist
    with _maybe_open(pathOrFile, 'rb') as fp:
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/plistlib.py", line 82, in _maybe_open
    with open(pathOrFile, mode) as fp:
FileNotFoundError: [Errno 2] No such file or directory: '/builds/scriptworker/work/0/Firefox Nightly.app/Contents/Frameworks/ChannelPrefs.framework/Contents/Info.plist'
```

TLDR; we can't sign the binaries like we do on bundles, we have to sign the folder, like `codesign [...] ChannelPrefs.framework/`

Comment (bhearsum) [reply iscript/src/iscript/mac.py:305]: OK, clearly I misunderstood what was in a framework!

**Review by bhearsum — APPROVED** [review]
